addons/trenches/functions/fnc_getSurfaceTexturePath.sqf

```diff
@@ -0,0 +1,66 @@
+#include "script_component.hpp"
+/*
+* Author: chris579, Salbei
+* Gets the path to the ground texture under the object
+*
+* Arguments:
+* 0: trench <OBJECT>
+*
+* Return Value:
+* Surface path <STRING>
+*
+* Example:
+* [TrenchObj] call ace_trenches_fnc_getSurfaceTexturePath
+*
+* Public: No
+*/
+
+params [
+    ["_object", objnull, [objNull]]
+];
+
+if (isNull _object) exitWith {};
+if !((isNil QGVAR(automaticFileSetup)) && (isText (configFile >> "CfgWorlds" >> worldName >> "surfaceTextureBasePath"))) exitWith {DEFAULT_TEXTURE};
+
+private _surfaceType = surfaceType (position _object);
+
+private _getTexturePath = {
+    params["_surfaceType", "_basePath", "_filePrefix"];
+
+    // remove leading #
+    private _parsedSurfaceType = _surfaceType select [1, count _surfaceType];
+    // check for overridden surface paths
+    private _overrideCfg = configFile >> "CfgWorlds" >> worldName >> "Surfaces" >> _parsedSurfaceType >> "texturePath";
+    if (isText (_overrideCfg)) exitWith {
+        getText(_overrideCfg)
+    };
+    // get config file wildcard
+    private _fileWildcard = getText(configfile >> "CfgSurfaces" >> _parsedSurfaceType >> "files");
+    // remove * in file wildcard
+    private _fileNameArr = _fileWildcard splitString "";
+    if (_fileNameArr find "*" > -1) then {
+        _fileNameArr deleteAt (_fileNameArr find "*");
+    };
+
+    format["%1\%2%3", _basePath, (_fileNameArr joinString ""), _filePrefix];
+};
+
+private _result = [];
```

Comment:
This could be a 
_result = if () then {} else {};

Reply:
I do not understand why this could be an if, in that line multiple string variables are merged. there is no else case.

Reply:
` there is no else case` 
Wat? Line 53? 
https://github.com/acemod/ACE3/pull/6712/files/08ff4366fcf5dfccef41a7b6eba594faa967cc99#r238088539
I can see it. It's right there.

Both cases set _result. Instead of doing that just return it and move the whole thing into _result. Exactly like I wrote.

Reply:
Ah sry, looked at the wrong line.